src/lib/builders/dialog/create.ts

```diff
@@ -228,17 +206,16 @@ export function createDialog(props?: CreateDialogProps) {
 				}),
 
 				effect([closeOnOutsideClick, open], ([$closeOnOutsideClick, $open]) => {
-					return useClickOutside(node, {
-						enabled: $open,
-						handler: (e: PointerEvent) => {
+					return useModal(node, {
```

Comment:
Could we maybe have it receive the stores directly? That way we don't need to do an effect here.

Reply:
So you're thinking we pass in the stores and just move the `effect`s into the `useModal`? I think that's a good idea and would clean this up quite a bit.

Reply:
Consolidated the effect into the action